webapp/cube/views.py

```diff
@@ -50,15 +55,17 @@ def cube_microcerts():
             badge = COURSE_DATA[course_id].get("logo")
             topics = COURSE_DATA[course_id].get("topics")
 
-            grade = None
-            try:
-                grade = cube_api.get_course_grades(course_id, username)[0]
-            except KeyError:
-                pass
-
-            status = "not-enrolled"
-            if grade:
-                status = "passed" if grade["passed"] else "enrolled"
+            if course_id in enrollments:
+                status = "enrolled"
+                try:
+                    grade = cube_api.get_course_grades(course_id, username)[0]
+                    if grade["passed"] and grade["username"] == username:
+                        status = "passed"
+                except KeyError:
+                    # There is no grade the user is just "enrolled"
+                    pass
```

Comment:
There is no need to put
```python
grade = cube_api.get_course_grades(course_id, username)[0]
```
inside this try block, right?

Also can we not change the condition to avoid the try/except?

Reply:
Update do `IndexError` so it probably makes more sense now :smile:

Reply:
I see, okay!